lib/WebService/GrowthBook.pm

```diff
@@ -0,0 +1,172 @@
+package WebService::GrowthBook;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+no indirect;
+use feature qw(state);
+use Object::Pad;
+use JSON::MaybeUTF8 qw(decode_json_text);
+use Scalar::Util qw(blessed);
+use Log::Any qw($log);
+use WebService::GrowthBook::FeatureRepository;
+use WebService::GrowthBook::Feature;
+use WebService::GrowthBook::FeatureResult;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::GrowthBook - sdk of growthbook
+
+=head1 SYNOPSIS
+
+    use WebService::GrowthBook;
+    my $instance = WebService::GrowthBook->new(client_key => 'my key');
+    $instance->load_features;
+    if($instance->is_on('feature_name')){
+        # do something
+    }
+    else {
+        # do something else
+    }
+    my $string_feature = $instance->get_feature_value('string_feature');
+    my $number_feature = $instance->get_feature_value('number_feature');
+    # get decoded json
+    my $json_feature = $instance->get_feature_value('json_feature');
+
+=head1 DESCRIPTION
+
+    This module is a sdk of growthbook, it provides a simple way to use growthbook features.
+
+=cut
+
+# singletons
+my  $feature_repository = WebService::GrowthBook::FeatureRepository->new;
+
+class WebService::GrowthBook {
+    field $url: param //= 'https://api.growthbook.io/api/v1';
+    field $client_key: param //= '';
+    field $features: param //= {};
+
+    method load_features {
+        if(!$client_key) {
+            die "Must specify 'client_key' to refresh features";
+        }
```

Comment:
if we dont specify a default value for the client_key param. The object creation itself would fail and would let us remove this check. What are your thoughts ?

Reply:
what I considered (or I guess python version considered) is that we can set features when we create Growthbook, like this:
```
my $gb = GrowthBook->new();
$gb->set_features(features => [{id => 'string-feature', default_value => 'abc'});

```

But now I think you are right. we needn't consider such case now.